test/common/thread/lock_guard_test.cc

```diff
@@ -0,0 +1,27 @@
+#include "common/common/thread.h"
+
+#include "gtest/gtest.h"
+#include "library/common/thread/lock_guard.h"
+
+namespace Envoy {
+namespace Thread {
+
+class ThreadTest : public testing::Test {
+protected:
+  ThreadTest() = default;
+  int a_ ABSL_GUARDED_BY(a_mutex_){0};
+  MutexBasicLockable a_mutex_;
+  int b_{0};
+};
+
+TEST_F(ThreadTest, TestOptionalReleasableLockGuard) {
+  OptionalReleasableLockGuard lock(nullptr);
+  EXPECT_EQ(1, ++b_);
+
+  OptionalReleasableLockGuard lock2(&a_mutex_);
+  EXPECT_EQ(1, ++a_);
+  lock2.release();
+}
```

Comment:
Should we add a test to validate that the [mutex unlocks on dealloc](https://github.com/lyft/envoy-mobile/pull/616/commits/f3d8ff27efe6035b122e65b35ae4bdd23c44b8d6#diff-c9f5b5ee47e72ae0286699c9caffd68dR34)?

Reply:
the first lock guard is testing that. If it did not. The test would not compile thanks to the thread annotations.